### PR TITLE
Java API: Documentation cleanup

### LINKED ISSUE #4417: Java API documentation has broken links
**Describe the bug**
Java API descrition document: https://github.com/microsoft/onnxruntime/blob/master/docs/Java_API.md refers to broken link: https://github.com/microsoft/onnxruntime/blob/master/java/sample/ScoreMNIST.java

**Urgency**
Not urgent

### PR DESCRIPTION
**Pull request by yuzawa-san**
**Description**: The Java docs (the markdown, not the javadoc) are little stale and need to be freshened up

**Motivation and Context**
- The Java API is now published to maven. This is big deal (at least I think so), so I moved that to the top of docs.
- I tried to make the `docs/Java_API.md` more about using the library as a dependency, how to use, and provide example code to bring it in line with the other languages' documentation.
- I tried to make `java/README.md` more about building, testing, and developing locally.
- The two docs redirect users with differing intent to the opposite documentation and vice versa.

Fixes #4417

### CODE REVIEW

File: docs/Java_API.md [hunk 1]
```diff
@@ -1,10 +1,35 @@
 # ONNX Runtime Java API
 The ONNX runtime provides a Java binding for running inference on ONNX models on a JVM, using Java 8 or newer.
 
-Two jar files are created during the build process, one contains the onnxruntime shared library, the JNI binding and the Java class files, and the other only contains the class files. By default the shared libraries are loaded from the classpath in a folder called `/lib`, if you wish to have them load from `java.library.path` then supply `-DORT_LOAD_FROM_LIBRARY_PATH` to the JVM at runtime.
+Release artifacts are published to Maven Central for use as a dependency in most Java build tools:
+
+* [com.microsoft.onnxruntime:onnxruntime](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime) - cross-platform CPU artifact
+* [com.microsoft.onnxruntime:onnxruntime_gpu](https://search.maven.org/artifact/com.microsoft.onnxruntime/onnxruntime_gpu) - cross-platform GPU (CUDA) artifact
+
+These artifacts are currently built for common OS/Architecture combinations: Windows x64, Linux x64, MacOS x64.
```
Comment (Craigacp) [inline docs/Java_API.md:9]: I think the GPU binary is only built for Windows & Linux, as macOS doesn't support CUDA anymore. Also it's `macOS` not `MacOS`.

File: docs/Java_API.md [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+
+For building locally, please see the [Java API development documentation](../java/README.md) for more details.
+
+For customization of the loading mechanism of the shared library, please see [advanced loading instructions](../java/README.md#advanced-loading).
+
+## API Reference
+
+The Javadoc is available [here](https://microsoft.github.io/onnxruntime/java/index.html).
```
Comment (Craigacp) [inline docs/Java_API.md:17]: @faxu we should update with the Javadoc for v1.3.1, as this is still the older Javadoc.

Comment (yuzawa-san) [reply docs/Java_API.md]: I found this https://stackoverflow.com/a/58465995

We publish the javadoc to maven alongside our artifacts, so this site basically takes the javadoc from there and displays it in a CDN-backed webpage for free. This way we do not have to keep upgrading the gh-pages branch. Maybe the old /java/index.html will need to redirect for posterity.

https://javadoc.io/doc/com.microsoft.onnxruntime/onnxruntime

It also handles different versions which is nice.

Is this a reasonable solution?

Alternatively, could the javadoc be built and pushed to the gh-pages branch using the azure CI process?

**Comment by Craigacp** [comment]
Might be worth updating the version stamp on the main README.md from `8-13` to `8-14` as I've tested it on 14 and it works just fine. Of course, 15 is out in September and it'll work just fine with that too, so maybe `8+` is sufficient?

File: docs/Java_API.md (hunk 2 above)
Comment (Craigacp) [reply docs/Java_API.md]: I agree that seems reasonable, but it's up to MS if they want to host the docs themselves or push it off to some third party.

I think building the javadoc as part of the gh-pages is reasonable, I don't recall how they got built initially.

File: docs/Java_API.md [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
+For customization of the loading mechanism of the shared library, please see [advanced loading instructions](../java/README.md#advanced-loading).
+
+## API Reference
+
+The Javadoc is available [here](https://javadoc.io/doc/com.microsoft.onnxruntime/onnxruntime).
 
 ## Sample Code
 
+An example implementation is located in
+[src/test/java/sample/ScoreMNIST.java](./src/test/java/sample/ScoreMNIST.java).
```
Comment (askhade) [inline docs/Java_API.md:24]: This link does not work it should be "../java/src/test/java/sample/ScoreMNIST.java" instead

**Comment by yuzawa-san** [comment]
Is there anything else outstanding related with these changes?

**Pull request merged**